### PR TITLE
docs: update Users docs for auto-login removal [DET-2992]

### PR DESCRIPTION
**Pull request by justin-determined-ai**
## Test Plan

Compiled and checked the docs locally.

## Commentary (optional)

A bit of overlap with the rest of the documentation, but this is intended to give as much necessary information to get started as possible.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/topic-guides/users.txt [hunk 1]
```diff
@@ -3,56 +3,34 @@
...
-Getting Started
----------------
+.. code::
 
-To start using users in a Determined cluster, the system administrator should
-first set a password for the ``admin`` user as described above.
+   det -u admin user change-password
 
-Next, the ``admin`` user should deactivate the ``determined`` user:
+The ``determined`` user is designed for ease of use on a single-user installation. However, teams will benefit from creating a Determined user account for each individual. Namely, having individual user accounts will provide clearer authorship and improved filtering of entities. To create users in a Determined installation, the ``admin`` user should be used to create user accounts for each individual requiring access to the installation:
```
Comment (neilconway) [inline docs/topic-guides/users.txt:21]: non-blocking: "requiring access to the installation" -> "that would like to use Determined"?

File: docs/topic-guides/users.txt [hunk 2]
```diff
@@ -3,56 +3,34 @@
...
-would like to access Determined:
+Then, the ``determined`` user can be deactivated:
 
-::
+.. code::
 
-   det -u admin user create <username>
+   det -u admin user deactivate determined
+
+Any entities that were created by this user prior to deactivation will remain.
```
Comment (neilconway) [inline docs/topic-guides/users.txt:33]: non-blocking: I thought the previous text was good. "This ensures that no one can access the Determined cluster as the ``determined`` user (any objects that were created by this user will remain)."

File: docs/topic-guides/users.txt (hunk 1 above)
Comment (neilconway) [reply docs/topic-guides/users.txt:21]: non-blocking: "will provide clearer" -> "provides clearer"

File: docs/topic-guides/users.txt [hunk 3]
```diff
@@ -3,56 +3,34 @@
...
+Determined is designed for teams of machine learning developers. Any entity (such as an experiment or a notebook) is visible to all users on the same installation of Determined, regardless of who created it.
 
-.. note::
+Getting Started
+---------------
 
-   The ``admin`` user initially has a blank password. The ``user
-   change-password`` CLI command can be used by the admin to update this
-   password:
+Each Determined installation comes with two user accounts: ``admin`` and ``determined``. Both of these accounts have blank passwords by default.
```
Comment (neilconway) [inline docs/topic-guides/users.txt:11]: non-blocking: "Initially, a Determined installation has two user accounts: ``admin`` and ``determined``."

Comment (justin-determined-ai) [reply docs/topic-guides/users.txt:11]: Keep the language about blank passwords?

**Pull request merged**